### PR TITLE
docs: Update library package markdown-magic transformations to split "header" and "footer" details

### PR DESCRIPTION
**Pull request by Josmithr**
The existence of a singular "library package README" transformation encourages developers to not include other necessary details in package READMEs. It was also previously unused in a lot of places because it was difficult to interleve the common details with custom ones.

This PR removes that template and introduces 2 new ones: a "header" template and a "footer" template, with recommended guidance for where each should be used.

Also updates existing consumers of the removed template to migrate to the new ones.

### CODE REVIEW

File: packages/loader/test-loader-utils/README.md [hunk 1]
```diff
@@ -2,14 +2,43 @@
 
 Test utilities for the Fluid Framework Loader. Includes mock implementations of various loader interfaces.
 
-<!-- AUTO-GENERATED-CONTENT:START (LIBRARY_PACKAGE_README:scripts=FALSE&installation=FALSE&apiDocs=FALSE&help=FALSE) -->
+<!-- AUTO-GENERATED-CONTENT:START (LIBRARY_PACKAGE_README_HEADER&apiDocs=FALSE) -->
```
Comment (Josmithr) [inline packages/loader/test-loader-utils/README.md:5]: Future improvement plan: Default `apiDocs` based on package scope. Disabled (by default) for `@fluid-internal` and `@fluid-private`

File: packages/loader/test-loader-utils/README.md [hunk 2]
```diff
@@ -2,14 +2,39 @@
...
+
+```bash
+npm i @fluid-private/test-loader-utils
+```
+
+<!-- prettier-ignore-end -->
+
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+<!-- AUTO-GENERATED-CONTENT:START (LIBRARY_PACKAGE_README_FOOTER&apiDocs=FALSE) -->
```
Comment (Josmithr) [inline packages/loader/test-loader-utils/README.md:33]: Future improvement plan: Default `apiDocs` based on package scope. Disabled (by default) for `@fluid-internal` and `@fluid-private`

File: packages/loader/test-loader-utils/README.md [hunk 3]
```diff
@@ -2,14 +2,39 @@
...
 
+## Using Fluid Framework libraries
+
+When taking a dependency on a Fluid Framework library's public APIs, we recommend using a `^` (caret) version range, such as `^1.3.4`.
+While Fluid Framework libraries may use different ranges with interdependencies between other Fluid Framework libraries,
+library consumers should always prefer `^`.
+
+If using any of Fluid Framework's unstable APIs (for example, its `beta` APIs), we recommend using a more constrained version range, such as `~`.
+
+## Installation
```
Comment (Josmithr) [inline packages/loader/test-loader-utils/README.md:21]: Future improvement plan: Don't include installation instructions for `@fluid-private` and `@fluid-internal` packages.

File: tools/markdown-magic/src/md-magic.config.cjs [hunk 4]
```diff
@@ -106,19 +106,15 @@ const generateTrademarkSection = (headingOptions) =>
 	});
```
Comment (Josmithr) [inline tools/markdown-magic/src/md-magic.config.cjs:106]: Plans for future improvements (in subsequent PRs):
1. By default, don't include the following sections for `@fluid-private` and `@fluid-internal` packages:
    - Installation instructions
    - API docs links
2. By default, configure the installation instructions section's "devDependency" argument based on whether or not the package is in the `@fluid-tools` scope (can still be overridden)

File: tools/markdown-magic/src/md-magic.config.cjs [hunk 5]
```diff
@@ -208,7 +170,7 @@ function libraryPackageReadmeTransform(content, options, config) {
 		sections.push(generatePackageScriptsSection(scriptsTable, sectionHeadingOptions));
 	}
 
-	if (options.contributionGuidelines !== "FALSE") {
+	if (options.clientRequirements !== "FALSE") {
```
Comment (Josmithr) [inline tools/markdown-magic/src/md-magic.config.cjs:173]: Bug I detected by leveraging the new groupings.

File: azure/packages/azure-local-service/README.md [hunk 6]
```diff
@@ -45,11 +53,59 @@ $env:PORT=6502
...
 
-<!-- AUTO-GENERATED-CONTENT:START (README_CONTRIBUTION_GUIDELINES_SECTION:includeHeading=TRUE) -->
+<!-- AUTO-GENERATED-CONTENT:START (LIBRARY_PACKAGE_README_FOOTER) -->
 
 <!-- prettier-ignore-start -->
 <!-- NOTE: This section is automatically generated using @fluid-tools/markdown-magic. Do not update these generated contents directly. -->
 
+## API Documentation
+
+API documentation for **@fluidframework/azure-local-service** is available at <https://fluidframework.com/docs/apis/azure-local-service>.
```
Comment (alexvy86) [inline azure/packages/azure-local-service/README.md:63]: This link 404s 🤔

File: tools/markdown-magic/README.md [hunk 7]
```diff
@@ -91,6 +94,25 @@ Arguments:
...
+
+#### `LIBRARY_PACKAGE_README_FOOTER`
+
+Generates simple "footer" contents for a library package README.
+
+Generally recommended for inclusion at the end of the README.
+
+Notes:
+
+-   This strictly intended as a starter template to remove the need for some handwritten boilerplate.
```
Comment (alexvy86) [inline tools/markdown-magic/README.md:106]:
```suggestion
-   This is strictly intended as a starter template to remove the need for some handwritten boilerplate.
```

File: tools/markdown-magic/README.md [hunk 8]
```diff
@@ -91,6 +94,25 @@ Arguments:
...
+    You will still need to fill in semantic and usage information.
+-   This is effectively just a wrapper around lower-level templates.
```
Comment (alexvy86) [inline tools/markdown-magic/README.md:108]: Might be nice to say which templates? So if someone wants to go from this to something custom but that still looks close to it, they know which other templates they'll need.

**Review by alexvy86 — APPROVED** [review]
A couple of small things but I like it overall. Also the future plans. I'm a bit conflicted about what kind of instructions we should be providing in all the packages that have to do with the encapsulated model... the declarative model ones seem fine, but instructions in the others feels like we encourage people to install them, which I don't know how to feel about.

File: azure/packages/azure-local-service/README.md (hunk 6 above)
Comment (Josmithr) [reply azure/packages/azure-local-service/README.md:63]: Yeah, I noticed that too. But there _should_ be docs for that.
@zhenmichael any idea why these API docs don't appear to be getting published to the website?

**Comment by Josmithr** [comment]
> I'm a bit conflicted about what kind of instructions we should be providing in all the packages that have to do with the encapsulated model... the declarative model ones seem fine, but instructions in the others feels like we encourage people to install them, which I don't know how to feel about.

The READMEs are generally discovered in 2 ways:
1. By looking at the package on `npmjs`
1. By looking at the package directly in the GitHub

In either of these cases, the user is already aware of the package. In my mind, if they're already looking at it, there isn't harm in explaining how to use it.

Packages that are not actually intended for _any_ external use should be in one of our special namespace scopes (`@fluid-internal` or `@fluid-private`). In those cases, we can and should omit this information. I mentioned this in future work plans.

**Pull request merged**